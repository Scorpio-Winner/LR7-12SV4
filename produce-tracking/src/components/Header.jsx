import React from 'react';
import { AppBar, Toolbar, Typography } from '@mui/material';
import Navigation from './Navigation';

const Header = () => {
  return (
    <AppBar position="static" sx={{ backgroundColor: '#f5f5f5' }}>
      <Toolbar>
        <Typography variant="h6" component="div" sx={{ flexGrow: 1, color: 'black' }}>
          Produce Tracking
        </Typography>
        <Navigation />
      </Toolbar>
    </AppBar>
  );
}

export default Header;