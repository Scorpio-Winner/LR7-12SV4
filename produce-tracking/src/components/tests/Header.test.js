import React from 'react';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import Header from '../Header';

describe('Header', () => {
  it('renders correctly', () => {
    const { getByText } = render(
      <Router>
        <Header />
      </Router>
    );

    expect(getByText('Produce Tracking')).toBeTruthy();
  });

  it('matches the snapshot', () => {
    const tree = renderer
      .create(
        <Router>
          <Header />
        </Router>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
