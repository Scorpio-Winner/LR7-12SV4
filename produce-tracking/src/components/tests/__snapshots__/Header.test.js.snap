// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header matches the snapshot 1`] = `
<header
  className="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionStatic css-4um4w8-MuiPaper-root-MuiAppBar-root"
>
  <div
    className="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-hyum1k-MuiToolbar-root"
  >
    <div
      className="MuiTypography-root MuiTypography-h6 css-piwjgx-MuiTypography-root"
    >
      Produce Tracking
    </div>
    <div
      className="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-hyum1k-MuiToolbar-root"
    >
      <div
        className="MuiBox-root css-i9gxme"
      >
        <a
          className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-1n54ss-MuiTypography-root-MuiLink-root"
          href="/orders"
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
        >
          All orders
        </a>
        <a
          className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-1n54ss-MuiTypography-root-MuiLink-root"
          href="/report"
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
        >
          Report
        </a>
        <a
          className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-1n54ss-MuiTypography-root-MuiLink-root"
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
        >
          Logout
        </a>
      </div>
    </div>
  </div>
</header>
`;
